Source/WebCore/style/StyleResolver.cpp

```diff
@@ -716,6 +716,18 @@ void Resolver::applyMatchedProperties(State& state, const MatchResult& matchResu
         state.setUserAgentAppearanceStyle(WTFMove(userAgentStyle));
     }
 
+    // If there is a base appearance style, checkout if we need to apply base appearance styles.
+    bool shouldIncludeBaseAppearanceStyle = false;
+    if (matchResult.hasBaseAppearanceStyle) {
+        auto throwaway = RenderStyle::createPtr();
+        Builder builder(*throwaway, builderContext(state), matchResult, CascadeLevel::Author, includedProperties);
+        builder.applyProperty(CSSPropertyAppearance);
+        shouldIncludeBaseAppearanceStyle = throwaway->appearance() == StyleAppearance::Base;
+    }
+
+    if (shouldIncludeBaseAppearanceStyle)
+        includedProperties.add(PropertyCascade::PropertyType::BaseAppearanceStyle);
```

Comment:
The UserAgentAppearanceStyle stuff above is not needed in base case, or needs to take this into account right?